homeassistant/components/roborock/sensor.py

```diff
@@ -134,6 +139,34 @@ class RoborockSensorDescription(
         native_unit_of_measurement=PERCENTAGE,
         device_class=SensorDeviceClass.BATTERY,
     ),
+    # Only available on some newer models
+    RoborockSensorDescription(
+        key="clean_percent",
+        icon="mdi:progress-check",
+        translation_key="clean_percent",
+        value_fn=lambda data: data.status.clean_percent,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=PERCENTAGE,
+    ),
+    # Only available with more than just the basic dock
+    RoborockSensorDescription(
+        key="dock_error",
+        icon="mdi:garage-open",
+        translation_key="dock_error",
+        value_fn=lambda data: data.status.dock_error_status.name
+        if data.status.dock_type != RoborockDockTypeCode.no_dock
+        else None,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.ENUM,
+        options=RoborockDockErrorCode.keys(),
+    ),
+    RoborockSensorDescription(
```

Comment:
This is missing the unit of measurement

Reply:
Swore I had that, must have deleted it, thanks